src/clr-angular/forms/select/select.ts

```diff
@@ -4,24 +4,71 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import {AfterContentInit, Component, ContentChild, ElementRef, HostListener, ViewChild} from "@angular/core";
+import {isPlatformBrowser} from "@angular/common";
+import {
+    AfterContentInit,
+    Component,
+    ContentChild,
+    ElementRef,
+    HostListener,
+    PLATFORM_ID,
+    Renderer2,
+    ViewChild
+} from "@angular/core";
 
 import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
 import {IfOpenService} from "../../utils/conditional/if-open.service";
 import {TAB} from "../../utils/key-codes/key-codes";
+
 import {ClrOptions} from "./options";
+import {OptionSelectionService} from "./providers/option-selection.service";
+import {SelectDomAdapter} from "./utils/select-dom-adapter";
+import {SelectNoopDomAdapter} from "./utils/select-noop-dom-adapter";
+
+// Fixes build error
+// @dynamic (https://github.com/angular/angular/issues/19698#issuecomment-338340211)
+export const selectDomAdapterFactory = (platformId: Object) => {
+    if (isPlatformBrowser(platformId)) {
+        return new SelectDomAdapter();
+    } else {
+        return new SelectNoopDomAdapter();
+    }
+};
 
 @Component({
     selector: "clr-select",
     templateUrl: "select.html",
-    providers: [IfOpenService, {provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef}],
+    providers: [
+        IfOpenService, {provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef}, OptionSelectionService,
+        {provide: SelectDomAdapter, useFactory: selectDomAdapterFactory, deps: [PLATFORM_ID]}
+    ],
     host: {"[class.clr-select]": "true"}
 })
 export class ClrSelect implements AfterContentInit {
     @ViewChild("input") input: ElementRef;
     @ContentChild(ClrOptions) options: ClrOptions;
 
-    constructor(private ifOpenService: IfOpenService) {}
+    constructor(private ifOpenService: IfOpenService, private optionSelectionService: OptionSelectionService,
+                private renderer: Renderer2, private domAdapter: SelectDomAdapter) {
+        this.initializeSubscriptions();
+    }
+
+    private initializeSubscriptions(): void {
+        this.optionSelectionService.selectionChanged.subscribe(() => {
```

Comment:
Need to track this subscription and unsubscribe on destroy.

Reply:
Oh wow. Cant believe I missed this. Thanks!